### ISSUE TITLE
Adopting new IPC-7351C guidelines

### ISSUE THREAD

**Issue by shamil-ru** [qeda/qeda#14]
Changes are to be supported:
- Land pattern naming conventions
- New pad calculation rules
- Silkscreen updates
- Terminal leads to mechanical layer

**Comment by Gasman2014** [comment]
Will this enhancement also address the orientation of the generated footprints? The zero orientation (IPC7351 A or IEC 61188-7 B) should be selectable. Also the zero orientation should be horizontal for two pad components with pad 1 to the left. The generated footprints are presently vertical for two pad components and zero aligned step/vrml models need to be rotated to fit.

**Comment by Gasman2014** [comment]
![screen shot 2017-03-13 at 15 12 38](https://cloud.githubusercontent.com/assets/8401593/23863723/59d1d178-0808-11e7-80fc-f1c4536889c0.png)

Zero component orientation should be horizontal for two pin components in IPC7351A. Current output is vertical and hence components require rotation.

QEDA
![screen shot 2017-03-13 at 16 18 25](https://cloud.githubusercontent.com/assets/8401593/23863906/e1b82006-0808-11e7-809f-1cda1efb4c52.png)
![screen shot 2017-03-13 at 16 19 03](https://cloud.githubusercontent.com/assets/8401593/23863910/e2e06e70-0808-11e7-97bf-d528b6653300.png)


 IPC7351
![screen shot 2017-03-13 at 16 25 26](https://cloud.githubusercontent.com/assets/8401593/23864214/baa8d018-0809-11e7-99bb-1e4a965412cd.png)
![screen shot 2017-03-13 at 16 25 14](https://cloud.githubusercontent.com/assets/8401593/23864224/bf7ba962-0809-11e7-9487-ad23fe61df5c.png)

**Comment by syakupov** [comment]
Yes you are right. This is due to `bodyWidth` and `bodyLength` confusion. Now `bodyWidth` is always horizontal. Now I understand this was bad decision. I hope this issue will be fixed in v0.4. But unfortunately new version will be incompatible with old libraries.

I will think about customizable zero position in new `ipc-7351c` style.

**Comment by shamil-ru** [comment]
Yes you are right. This is due to `bodyWidth` and `bodyLength` confusion. Now `bodyWidth` is always horizontal. Now I understand this was bad decision. I hope this issue will be fixed in v0.4. But unfortunately new version will be incompatible with old libraries.

I will think about customizable zero position in new `ipc-7351c` style.

**Comment by Gasman2014** [comment]
I don’t think backward compatibility is a big issue. You will not be reading the libraries back into QEDA and the .yaml files don’t need to change. You just need to add the ‘IPC_Zero position’ setting as an option (i.e. maintain backward compatibility OR adopt IPC zero rotation) and then generate the footprints horizontally not vertically. Think this is only an issue for 2-part components in QEDA anyhow? The whole rotation thing is quite confusing esp the EiA481 / vs IPC.

John

**Comment by dvc94ch** [comment]
Any advice on how to get a copy of ipc-7351c? I can't find a purchase link anywhere... Has it even been released yet officially?

**Comment by Gasman2014** [comment]
Aparantly progress seems to have stalled. https://www.pcblibraries.com/forum/ipc7351c-draft-or-release-date_topic1818.html

According to this, the amended solder joint goals in the documents section have been updated to include all the agreed changes so should be c compliant. 

You have to register but it is free.

**Comment by Houkime** [comment]
So... if IPC is effectively dead...
It seems like oshw community should produce guidelines of their own now and compile stuff from PCB manufacturers, datasheets and experience?

Is there a repo for that already?
